openmc/model/surface_composite.py

```diff
@@ -0,0 +1,394 @@
+    cyl : openmc.Cylinder
+        Underlying cylinder surface
+    bottom : openmc.Plane
+        Bottom planar surface of the cylinder
+    top : openmc.Plane
+        Top planar surface of the cylinder
+
+    """
+    _surface_names = ('cyl', 'bottom', 'top')
+
+    def __init__(self, center_base, height, radius, axis='z', **kwargs):
+        cx, cy, cz = center_base
+        if axis == 'x':
+            self.cyl = openmc.XCylinder(y0=cy, z0=cz, r=radius, **kwargs)
+            self.bottom = openmc.XPlane(x0=cx, **kwargs)
+            self.top = openmc.XPlane(x0=cx + height, **kwargs)
+        elif axis == 'y':
+            self.cyl = openmc.YCylinder(x0=cx, z0=cz, r=radius, **kwargs)
+            self.bottom = openmc.YPlane(y0=cy, **kwargs)
+            self.top = openmc.YPlane(y0=cy + height, **kwargs)
+        elif axis == 'z':
+            self.cyl = openmc.ZCylinder(x0=cx, y0=cy, r=radius, **kwargs)
+            self.bottom = openmc.ZPlane(z0=cz, **kwargs)
+            self.top = openmc.ZPlane(z0=cz + height, **kwargs)
+
+    def __neg__(self):
+        return -self.cyl & +self.bottom & -self.top
+
+    def __pos__(self):
+        return +self.cyl | -self.bottom | +self.top
+
+
+class RectangularParallelepiped(CompositeSurface):
+    """Rectangular parallelpiped composite surface
+
+    A rectangular parallelpiped is composed of six planar surfaces. This class
+    acts as a proper surface, meaning that unary `+` and `-` operators applied
+    to it will produce a half-space. The negative side is defined to be the
+    region inside of the rectangular parallelpiped.
+
+    .. versionadded:: 0.12
+
+    Parameters
+    ----------
+    xmin, xmax : float
+        Minimum and maximum x coordinates of the parallelepiped
+    ymin, ymax : float
+        Minimum and maximum y coordinates of the parallelepiped
+    zmin, zmax : float
+        Minimum and maximum z coordinates of the parallelepiped
+    **kwargs
+        Keyword arguments passed to underlying plane classes
+
+    Attributes
+    ----------
+    xmin, xmax : openmc.XPlane
+        Sides of the parallelepiped
+    ymin, ymax : openmc.YPlane
+        Sides of the parallelepiped
+    zmin, zmax : openmc.ZPlane
+        Sides of the parallelepiped
+
+    """
+    _surface_names = ('xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax')
+
+    def __init__(self, xmin, xmax, ymin, ymax, zmin, zmax, **kwargs):
+        self.xmin = openmc.XPlane(x0=xmin, **kwargs)
+        self.xmax = openmc.XPlane(x0=xmax, **kwargs)
+        self.ymin = openmc.YPlane(y0=ymin, **kwargs)
+        self.ymax = openmc.YPlane(y0=ymax, **kwargs)
+        self.zmin = openmc.ZPlane(z0=zmin, **kwargs)
+        self.zmax = openmc.ZPlane(z0=zmax, **kwargs)
+
+    def __neg__(self):
+        return +self.xmin & -self.xmax & +self.ymin & -self.ymax & +self.zmin & -self.zmax
+
+    def __pos__(self):
+        return -self.xmin | +self.ymax | -self.ymin | +self.ymax | -self.zmin | +self.zmax
+
+
+class Box(CompositeSurface):
```

Comment:
Yes, this class is meant to eventually mimic the `box` macrobody from MCNP. To provide a little more context, the original reason I wrote these classes is for automated conversion of MCNP models. So far, the only cases I've come across of MCNP models using the `box` macrobody were cases where the vectors _were_ axis-aligned (i.e., they could have used `rpp`). Perhaps it might be better to just remove `Box` for now since it's not really providing anything extra?

Reply:
Ah ok. Flipping through the MCNP manual I see what you're saying. I'd say leave `Box` out for now.